bin/custom/concerttours/src/concerttours/facades/impl/DefaultBandFacade.java

```diff
@@ -0,0 +1,60 @@
+package concerttours.facades.impl;
+
+import concerttours.data.BandData;
+import concerttours.enums.MusicType;
+import concerttours.facades.BandFacade;
+import concerttours.model.BandModel;
+import concerttours.service.BandService;
+import de.hybris.platform.servicelayer.dto.converter.Converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DefaultBandFacade implements BandFacade
+{
+    private BandService bandService;
+
+    private Converter<BandModel, BandData> bandConverter;
+
+    public void setBandService(final BandService bandService)
+    {
+        this.bandService = bandService;
+    }
+
+    public void setBandConverter(Converter<BandModel, BandData> bandConverter) {
+        this.bandConverter = bandConverter;
+    }
+
+    @Override
+    public List<BandData> getBands()
+    {
+        return bandService.getBands().stream()
+                .map(bandModel -> bandConverter.convert(bandModel))
+                .collect(Collectors.toList());
+    }
+    @Override
+    public BandData getBand(final String name)
+    {
+        if (name == null)
```

Comment:
Please consider using Objects.requireNonNull instead of manual checks
(or get rid of checks like this one at all, it will anyway throw a descriptive NPE when null is passed)